src/sys/mman.rs

```diff
@@ -258,6 +258,27 @@ pub unsafe fn madvise(addr: *mut c_void, length: size_t, advise: MmapAdvise) ->
     Errno::result(libc::madvise(addr, length, advise as i32)).map(drop)
 }
 
+/// Calls to `mprotect` are inherently unsafe, as changes to memory protections can lead to
+/// SIGSEGVs.
+///
+/// ```
+/// use nix::sys::mman::{mmap, mprotect, MapFlags, ProtFlags};
+/// use std::ptr;
+///
+/// let mut slice: &mut [u8] = unsafe {
+///     let mem = mmap(ptr::null_mut(), 1024, ProtFlags::PROT_NONE,
+///                    MapFlags::MAP_ANONYMOUS | MapFlags::MAP_PRIVATE, -1, 0).unwrap();
```

Comment:
`MAP_ANONYMOUS` is a nonstandard flag, so it doesn't make a great example.

Reply:
I've only ever used `mmap` with `MAP_ANONYMOUS`, but from what I can tell by the POSIX manpage, it would need a file descriptor if that flag isn't used. This has to either be backed by a real file or a typed memory object, the API for which `nix` doesn't support. This is starting to feel _way_ too complicated for a 3 line wrapper function, and would be more appropriate for a larger `mman` test suite.

Reply:
It looks like `MAP_ANON` at least passes on all the CI platforms, even though it's also not in POSIX. Good enough?